app/models/ep_postmaster/mailgun_post.rb

```diff
@@ -4,10 +4,20 @@ module EpPostmaster
   class MailgunPost
 
     VALID_EVENTS = %w(failed rejected).freeze
+    VALID_DOMAINS = [
+      "relay.360members.com",
+      "relay.staging.360members.com",
+      "relay.360unite.com",
+      "relay.staging.360unite.com",
+      "email.church360.org",
+      "email.staging.church360.org"
+    ].freeze
 
-    attr_accessor :message_id, :code, :message_headers, :error, :event, :recipient, :reply_to, :subject, :signature, :timestamp, :token, :from, :sender, :reason
+    attr_accessor :storage_url, :message_id, :code, :message_headers, :error, :event, :recipient, :reply_to, :subject, :signature, :timestamp, :token, :from, :sender, :reason
 
     def initialize(params)
+      storage_data = params["event-data"]["storage"] || {}
+      @storage_url = storage_data["url"]
```

Comment:
What's this used for?

Reply:
It's possible we don't get header data in the event data.  the storage url we do get.  And it contains the domain as part of the url. Which I use in the api_key lambda to determine which api_key to use, the one for 360members or church360.

Reply:
I was thinking it's wild that we have to get the domain information out of the storage URL ... but looking at the example event payloads ... that may be the only place that indicates it. 😅 It just feels so fragile ... like it's an incidental piece of data that we're relying on to determine where this is coming from. 😬 

I wonder if what Mailgun wants you do do/expects you to do would be to get the domain from the request itself... 🤔 Maybe we instead add the domain as a parameter, and then we can get it from the controller when it builds the `MailgunPost`:

```ruby
# mailgun_hooks_controller.rb

...

def authenticate_request!
  @mailgun_post = MailgunPost.new(params, domain: request.host)
  ...
end

# mailgun_post.rb

attr_reader :domain

def initialize(params, domain: nil)
  @domain = domain
  ...
end
```

This would have far less chance of breaking on the off-chance that Mailgun updates their API to omit the storage URL ... or changes their URL scheme entirely! 😱

Reply:
1). I'm not sure what request.host will be.
2) APIs change.  Any part of the API could change that could break something other than this.  3) We have a hard deadline and I need to move on to the current milestone.  I can revisit this and refactor it during the next refactor week to see if request.host will work.

Reply:
No need to get short. 😕 

I see your point and agree it's not time to bike shed this. The overall API -- using the lambda to allow the consuming app to examine the message and supply the correct Mailgun API key -- is sound and should allow for additional refactoring as needed.

  I'm okay merging this as-is, but using an unrelated part of the API to divine the domain _is_ a code smell, and we should address it sooner than later. With a bit of DNS configuration, I think the `request` method will probably be the way to go -- having each domain in Mailgun actually use the domain it's representing for its webhook. Just so long as we don't let it slip through the cracks and never circle back around to it... 😖